controllers/storageconsumer/storageconsumer_controller.go

```diff
@@ -721,6 +729,95 @@ func (r *StorageConsumerReconciler) reconcileNoobaaAccount() error {
 	return nil
 }
 
+func (r *StorageConsumerReconciler) deletionPhase() (reconcile.Result, error) {
```

Comment:
The structure of the function does not make a lot of sense. 
A better approach would be to have a set of checks for allowing the deletion as a separate function which then is invoked outside of this function to decide if we need to delete or to reconcile. And this function should assume safe for deletion